api/base.proto

```diff
@@ -111,3 +111,25 @@ message DiscoveryResponse {
   string version_info = 1;
   repeated google.protobuf.Any resources = 2;
 }
+
+message ApiConfigSource {
+  // APIs may be fetched via either REST or gRPC.
+  enum ApiType {
+    REST = 0;
+    GRPC = 1;
+  }
+  ApiType api_type = 1;
+  // Multiple cluster names may be provided. If > 1 cluster is defined, clusters
+  // will be cycled through if any kind of failure occurs.
+  repeated string cluster_name = 2;
```

Comment:
Is cluster_name the right concept here or should it be an address? I might just be revealing my ignorance. :)